packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -43,7 +43,10 @@ const VALID_REDIRECT_RESPONSE_CODES = new Set([301, 302, 303, 307, 308]);
 /**
  * Additional metadata for a server configuration route tree.
  */
-type ServerConfigRouteTreeAdditionalMetadata = Partial<ServerRoute>;
+type ServerConfigRouteTreeAdditionalMetadata = Partial<ServerRoute> & {
+  /** Indicates if the route has been matched with the Angular router routes. */
+  matched?: boolean;
```

Comment:
nit: I'd propose a more descriptive name for readability and easier maintenance:
```suggestion
  presentInClientRouter?: boolean;
```

Reply:
This is done during build time so I am less worried about that, if we set undefined or null, we’d need to filter it out so not to include it in the manifest